src/models/bucket.ts

```diff
@@ -41,8 +48,8 @@ async function updateInstanceCache(batch: number): Promise<void> {
 
 export interface ResultData {
   solver: string;
-  result: string;
-  graph: string;
+  result?: string;
+  graph?: string;
```

Comment:
Given they are either both null or both set, I'd slightly prefer if it could be a tuple? but this might not be worth the refactoring.

Reply:
Love it!